waspc/src/Wasp/AppSpec.hs

```diff
@@ -76,7 +77,7 @@ data AppSpec = AppSpec
     -- | Connection URL for a database used during development. If provided, generated app will
     -- make sure to use it when run in development mode.
     devDatabaseUrl :: Maybe String,
-    isCustomViteConfigUsed :: Bool
+    customViteConfigPath :: Maybe (Path' (Rel SourceExternalCodeDir) File')
```

Comment:
You made it relative to SourceExternalCodeDir -> that is correct, I am just wondering, do we care if it is in external code dir or not? Because we could be less specific and say it is in source project dir (so root of source project). Then it could be in ext src, or not, who knows. But ok, this is also fine on one hand, especially if you are relying on this property of it being in ext src.

Reply:
Either we add the `src` part in the AppSpec creation part or we add it in the generator 🤷 I think it's a bit arbitrary and maybe just keep it like this for now? I don't really any good reasons

Reply:
It's more about constraints -> is it important for us that custom vite config is in source external code dir? If so, then let's leave it as it is. If it doesn't matter, and we are ok with it being anywhere in the project, but for now we chose source external code dir, then I would go with path starting from the root of the project.